app/__main__.py

```diff
@@ -52,18 +75,94 @@ async def cancel(self, interaction: Interaction, button: Button) -> None:
         message = await interaction.original_response()
         await interaction.edit_original_response(embed=message.embeds[0], view=self)
 
-    async def _create_embed(self, interaction: Interaction) -> discord.Embed:
+    @discord.ui.button(
+        label="CPS Upgrade 10x", style=discord.ButtonStyle.gray, row=1, custom_id="upgradepersistent:cps10x"
+    )
+    async def cps_upgrade_ten(self, interaction: Interaction, button: Button) -> None:
+        """Upgrade CPS ten times."""
+        button.label = "CPS 10x Selected"
+        await interaction.response.defer()
+        new_profile, status_code = await self._upgrade_cps(interaction, 10)
+        await self._handle_status_code(interaction, status_code)
+        await interaction.client.database.update_profile(interaction.guild, interaction.user, new_profile)
+        await interaction.edit_original_response(embed=await self.create_embed(interaction), view=self)
+
+    @staticmethod
+    async def create_embed(interaction: Interaction) -> discord.Embed:
         """Create a custom embed to accompany the edited message upon upgrade."""
         profile: UserProfile = await interaction.client.database.get_profile(interaction.guild, interaction.user)
-        priority_cost = PRIOTIY_COST[profile.priority]
-        cps_cost = CPS_COST[profile.cps]
+        priority_cost: int = PRIORITY_COST[profile.priority]
+        cps_cost: int = CPS_COST[profile.cps]
         embed = discord.Embed(title="Upgrade menu", description="Select an upgrade to obtain")
         embed.add_field(
             name="Better CPS", value=f"Increase the amount of characters you can send per second\nCosts {cps_cost}"
         )
         embed.add_field(name="Higher Priority", value=f"Increase the priority of your messages\nCosts {priority_cost}")
         return embed
 
+    async def _upgrade_priority(self, interaction: Interaction) -> tuple[UserProfile, StatusCode]:
+        """Upgrade the priority of the user."""
+        profile: UserProfile = await interaction.client.database.get_profile(interaction.guild, interaction.user)
+        priority_cost: int = PRIORITY_COST[profile.priority]
+        if profile.coins < priority_cost:
+            return (profile, StatusCode.NOT_ENOUGH_COINS)
+        if profile.priority == MessagePriority.TOP:
+            return (profile, StatusCode.MAXIMUM_REACHED)
+        new_coins: int = profile.coins - priority_cost
+        new_priority: MessagePriority = PRIORITY_PIPELINE[PRIORITY_PIPELINE.index(profile.priority) + 1]
+        new_profile: UserProfile = UserProfile(coins=new_coins, priority=new_priority)
+        return (new_profile, StatusCode.SUCCESS)
```

Comment:
Could you remove some of the explicit type hints here? They're kinda distracting :( (and inference should get most or all of them)